backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/BulkDeleteUtil.java

```diff
@@ -0,0 +1,773 @@
+import org.apache.logging.log4j.Logger;
+import org.eclipse.sw360.common.utils.BackendUtils;
+import org.eclipse.sw360.datahandler.common.CommonUtils;
+import org.eclipse.sw360.datahandler.couchdb.AttachmentConnector;
+import org.eclipse.sw360.datahandler.entitlement.ComponentModerator;
+import org.eclipse.sw360.datahandler.entitlement.ReleaseModerator;
+import org.eclipse.sw360.datahandler.thrift.ReleaseRelationship;
+import org.eclipse.sw360.datahandler.thrift.SW360Exception;
+import org.eclipse.sw360.datahandler.thrift.Source;
+import org.eclipse.sw360.datahandler.thrift.changelogs.Operation;
+import org.eclipse.sw360.datahandler.thrift.components.Component;
+import org.eclipse.sw360.datahandler.thrift.components.BulkOperationNode;
+import org.eclipse.sw360.datahandler.thrift.components.BulkOperationNodeType;
+import org.eclipse.sw360.datahandler.thrift.components.BulkOperationResultState;
+import org.eclipse.sw360.datahandler.thrift.components.Release;
+import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.datahandler.permissions.PermissionUtils;
+
+import com.cloudant.client.api.model.Document;
+import com.cloudant.client.api.model.Response;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.eclipse.sw360.datahandler.common.SW360Assert.*;
+
+/**
+ * Provides a utility for the bulk delete function
+ */
+public class BulkDeleteUtil extends BackendUtils {
+
+    private static final String CONFLICT_ERROR = "conflict";
+    
+    private static final Logger log = LogManager.getLogger(BulkDeleteUtil.class);
+
+    private static final int LOOP_MAX = 100000;
+    
+    private ComponentDatabaseHandler componentDatabaseHandler;
+    private ComponentRepository componentRepository;
+    private ReleaseRepository releaseRepository;
+    private ProjectRepository projectRepository;
+    private ComponentModerator componentModerator;
+    @SuppressWarnings("unused")
+    private ReleaseModerator releaseModerator;
+    private AttachmentConnector attachmentConnector;
+    private AttachmentDatabaseHandler attachmentDatabaseHandler;
+    private DatabaseHandlerUtil dbHandlerUtil;    
+    
+    private BulkDeleteUtilInspector inspector;
+    
+    public interface BulkDeleteUtilInspector {
+        void checkVariables(Map<String, Release> releaseMap, Map<String, Component> componentMap,Map<String, Boolean> externalLinkMap, Map<String, List<String>> referencingReleaseIdsMap);
+        void checkLoopState(int loopCount, Map<String, Release> releaseMap, Map<String, Component> componentMap, Map<String, BulkOperationResultState> resultStateMap);
+        void checkleafReleaseIdsInLoop(int loopCount, Set<String> leafReleaseIds);
```

Comment:
change checkleafReleaseIdsInLoop to checkLeafReleaseIdsInLoop

Reply:
Done